### PR TITLE
Change Dataset Gallery cell type color to provide higher contrast

### PR DESCRIPTION
**Pull request by user27182**
### Overview

The color for the cell types is currently set to `dark`, but this color is static and does not change when the dark theme is used. This makes it hard to see.:
image

This PR changes it to the use the `surface` color from 
https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/styling.html#color-variables

### CODE REVIEW

File: doc/source/make_tables.py [hunk 1]
```diff
@@ -1455,12 +1455,15 @@ def fetch_multiblock(cls, kind: Literal["hetero", "homo", "single"]):
 class _BaseDatasetBadge:
     class SemanticColorEnum(StrEnum):
         """Enum of badge colors.
-        See: https://sphinx-design.readthedocs.io/en/latest/badges_buttons.html"""
+
+        See: https://sphinx-design.readthedocs.io/en/latest/badges_buttons.html
+        https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/styling.html#color-variables
```
Comment (tkoyama010) [inline doc/source/make_tables.py:1460]:
```suggestion
        See:
        - https://sphinx-design.readthedocs.io/en/latest/badges_buttons.html
        - https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/styling.html#color-variables
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by user27182** [comment]
Looks like not all the pydata colors are supported :(. It's probably strictly limited to just these: https://sphinx-design.readthedocs.io/en/latest/badges_buttons.html

**Comment by user27182** [comment]
Using the `muted` color in https://github.com/pyvista/pyvista/pull/6475/commits/1587c00bcf9811cc8efe95c0d372285c243f3ed4 works:
Screenshot 2024-08-08 at 11 36 30 AM
Screenshot 2024-08-08 at 11 36 39 AM

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**